pkg/client/camel/applyconfiguration/camel/v1/buildertask.go

```diff
@@ -96,3 +97,16 @@ func (b *BuilderTaskApplyConfiguration) WithBuildDir(value string) *BuilderTaskA
 	b.BuildDir = &value
 	return b
 }
+
+// WithSources adds the given value to the Sources field in the declarative configuration
+// and returns the receiver, so that objects can be build by chaining "With" function invocations.
+// If called multiple times, values provided by each call will be appended to the Sources field.
+func (b *BuilderTaskApplyConfiguration) WithSources(values ...*SourceSpecApplyConfiguration) *BuilderTaskApplyConfiguration {
+	for i := range values {
+		if values[i] == nil {
+			panic("nil value passed to WithSources")
```

Comment:
We should not panic but report the error and set the related CR status to error.

Reply:
This code has been generated, I have no idea how I can change its behavior but I will try to figure out

Reply:
We have the same behavior in the [integration spec](https://github.com/apache/camel-k/blob/main/pkg/client/camel/applyconfiguration/camel/v1/integrationspec.go#L63), so in practice, it would fail before. So if it is an issue, IMHO, it should be addressed in a dedicated ticket, don't you agree?

Reply:
okey, then likely it's something that should never happen. Please, feel free to proceed.